### DISCUSSION TITLE
NMS UI setup failed

### DISCUSSION THREAD

**Discussion by rholling** [magma/magma#11325] [Q&A]
Hi folks,

I'm stucking at the point, to setup the NMS UI. Until here, everything went well.
I followed the docs on magmacore.org and now: quick start guide, section "using the NMS UI":

the first two commands worked fine:
$ docker-compose build magmalte
$ docker-compose up -d

but with the next command I ran into trouble:
$ ./scripts/dev_setup.sh 

The output I've got is:
 
yarn run v1.22.5
$ node -r @fbcnms/babel-register scripts/migrate.js
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
2022-01-30T11:19:45.612Z [scripts/runMigrations.js] info: == 20190509000000-make-organization-unique: migrating =======
Failed to run migrations
Done in 14.01s.
yarn run v1.22.5
$ node -r '@fbcnms/babel-register' scripts/setPassword.js magma-test ad...@magma.test password1234
Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db

Why you should do it regularly:
https://github.com/browserslist/browserslist#browsers-data-updating
Creating a new user: email=ad...@magma.test, password=password1234
{ Error:
    at Query.run (../../../../lib/dialects/postgres/query.js:76:25)
    at retry (../../lib/sequelize.js:643:28)
    at process._tickCallback (internal/process/next_tick.js:68:7)
  name: 'SequelizeDatabaseError',
  parent:
   { error: column "tabs" does not exist
       at Parser.parseErrorMessage (../src/parser.ts:369:69)
       at Parser.handlePacket (../src/parser.ts:188:21)
       at Parser.parse (../src/parser.ts:103:30)
       at Socket.stream.on.buffer (../src/index.ts:7:48)
       at Socket.emit (events.js:198:13)
       at addChunk (_stream_readable.js:288:12)
       at readableAddChunk (_stream_readable.js:269:11)
       at Socket.Readable.push (_stream_readable.js:224:10)
       at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
     length: 97,
     name: 'error',
     severity: 'ERROR',
     code: '42703',
     detail: undefined,
     hint: undefined,
     position: '73',
     internalPosition: undefined,
     internalQuery: undefined,
     where: undefined,
     schema: undefined,
     table: undefined,
     column: undefined,
     dataType: undefined,
     constraint: undefined,
     file: 'parse_relation.c',
     line: '3194',
     routine: 'errorMissingColumn',
     sql:
      'SELECT "id", "email", "organization", "password", "role", "networkIDs", "tabs", "createdAt", "updatedAt" FROM "Users" AS "User" WHERE "User"."organization" = \'magma-test\' AND "User"."email" = \'ad...@magma.test\' LIMIT 1;',
     parameters: undefined },
  original:
   { error: column "tabs" does not exist
       at Parser.parseErrorMessage (../src/parser.ts:369:69)
       at Parser.handlePacket (../src/parser.ts:188:21)
       at Parser.parse (../src/parser.ts:103:30)
       at Socket.stream.on.buffer (../src/index.ts:7:48)
       at Socket.emit (events.js:198:13)
       at addChunk (_stream_readable.js:288:12)
       at readableAddChunk (_stream_readable.js:269:11)
       at Socket.Readable.push (_stream_readable.js:224:10)
       at TCP.onStreamRead [as onread] (internal/stream_base_commons.js:94:17)
     length: 97,
     name: 'error',
     severity: 'ERROR',
     code: '42703',
     detail: undefined,
     hint: undefined,
     position: '73',
     internalPosition: undefined,
     internalQuery: undefined,
     where: undefined,
     schema: undefined,
     table: undefined,
     column: undefined,
     dataType: undefined,
     constraint: undefined,
     file: 'parse_relation.c',
     line: '3194',
     routine: 'errorMissingColumn',
     sql:
      'SELECT "id", "email", "organization", "password", "role", "networkIDs", "tabs", "createdAt", "updatedAt" FROM "Users" AS "User" WHERE "User"."organization" = \'magma-test\' AND "User"."email" = \'ad...@magma.test\' LIMIT 1;',
     parameters: undefined },
  sql:
   'SELECT "id", "email", "organization", "password", "role", "networkIDs", "tabs", "createdAt", "updatedAt" FROM "Users" AS "User" WHERE "User"."organization" = \'magma-test\' AND "User"."email" = \'ad...@magma.test\' LIMIT 1;',
  parameters: {} }
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

I then followed the hint and run "npx browserslist@latest --update-db", with the result:

npx: Installierte 6 in 2.165s
Latest version:     1.0.30001304
Installed version:  1.0.30001304
caniuse-lite is up to date
caniuse-lite has been successfully updated

No target browser changes.

I re-run the command $ ./scripts/dev_setup.sh, but I've got the same result.

I'm running Ubuntu 20.04.3 LTS on a bare-metal.
Any help would be kind!

Thanks a lot
Robert

**Comment by sebathomas** [comment]
Looks like something went wrong with the DB migration script. It might help to remove the db volume and retry:
```bash
cd nms/packages/magmalte
docker-compose down -v
docker-compose up -d
./scripts/dev_setup.sh
```

**Reply by rholling8** [reply]
I tried it out, but I'm sorry, it didn't work.

The result/error is the same:

"Failed to run migrations"

**Reply by sebathomas** [reply]
It's hard to debug from the message unfortunately. I opened a PR to improve the logging, see #11379.

**Reply by sebathomas** [reply]
Possibly related bug report with a workaround: #11291

**Reply by rholling** [reply]
Thanks, the workaround #11291 works!
But I've no idea, if this could cause any problems in the future?
THX!

**Reply by sebathomas** [reply]
I don't think it should break anything (and I guess you can revert changes to the package.json after running the migrations successfully once).
